### PR TITLE
Ankit | Test | Prod - Blank page is getting displayed on tax reported in US company when tax number not present in company 

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cwfyw8u

### CODE REVIEW

File: apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html [hunk 1]
```diff
@@ -117,18 +117,20 @@
...
-        
+
                                     <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
                                     <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
                                     <tr mat-footer-row *matFooterRowDef="displayedColumns"></tr>
                                 </table>
                             </div>
                         </div>
                         <!-- No Data Found Message -->
-                        <div class="no-data" *ngIf="(accountWiseReport$ | async)?.responses?.length === 0 && !(isLoading$ | async) && !isTaxApiInProgress">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html:128]: why you are removing this condition ?

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html:128]: (accountWiseReport$ | async)?.responses?.length === 0  
This condition will be true if I have responses, but not responses if I don't have a tax number.
!(isLoading$ | async) 
this condition already checked by the parent div.
!isTaxApiInProgress
this condition already checked by the parent div

File: apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html [hunk 2]
```diff
@@ -117,18 +117,20 @@
...
-                            <h1 *ngIf="!getFormControl('taxNumber')?.value">{{ localeData?.no_sales_tax_available }}</h1>
-                            <h1 *ngIf="getFormControl('taxNumber')?.value">{{ commonLocaleData?.app_no_result_found }}</h1>
-                        </div>
+                        <ng-template #nodata>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html]:
```suggestion
                        <ng-template #noData>
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.html [hunk 3]
```diff
@@ -43,7 +43,7 @@
                         <giddh-page-loader></giddh-page-loader>
                     </div>
                     <div class=" mt-4" *ngIf="!(isLoading$ | async) && !isTaxApiInProgress">
-                        <div class="white-box w-100" *ngIf="(taxWiseReport$ | async)?.responses?.length">
+                        <div class="white-box w-100" *ngIf="(taxWiseReport$ | async)?.responses?.length; else nodata ">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.html]:
```suggestion
                        <div class="white-box w-100" *ngIf="(taxWiseReport$ | async)?.responses?.length; else noData ">
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.ts [hunk 4]
```diff
@@ -89,6 +90,22 @@ export class RateWiseReportComponent implements OnInit, OnDestroy {
...
+            if (response?.body?.length) {
+                this.getFormControl('taxNumber').patchValue(response.body[0]);
+            }
+        });
+    }
+/**
+* Loads the tax details of a company
+*
+* @private
+* @memberof VatLiabilitiesPayments
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.ts]:
```suggestion
* @memberof RateWiseReportComponent
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html [hunk 5]
```diff
@@ -38,7 +38,7 @@
                         <giddh-page-loader></giddh-page-loader>
                     </div>
                     <div class=" mt-4" *ngIf="!(isLoading$ | async) && !isTaxApiInProgress">
-                        <div class="white-box w-100" *ngIf="(taxAuthorityWiseReport$ | async)?.responses?.length">
+                        <div class="white-box w-100" *ngIf="(taxAuthorityWiseReport$ | async)?.responses?.length; else nodata">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html]:
```suggestion
                        <div class="white-box w-100" *ngIf="(taxAuthorityWiseReport$ | async)?.responses?.length; else noData">
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts [hunk 6]
```diff
@@ -82,6 +83,23 @@ export class TaxAuthorityReportComponent implements OnInit, OnDestroy {
...
+                this.getFormControl('taxNumber').patchValue(response.body[0]);
+            }
+        });
+    }
+
+    /**
+    * Loads the tax details of a company
+    *
+    * @private
+    * @memberof VatLiabilitiesPayments
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts]: correct this

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.html]: Done

File: apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.html (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.html]: Done

File: apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.ts]: Done

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html (hunk 5 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html]: Done

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts (hunk 6 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts]: done

**Review by dvCodeWorld — APPROVED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

**Pull request merged**